frontend/scripts/pipelinespec_pbjs.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Note: This script uses protobuf.js to generate pipeline_spec classes in Typescript.
+// Reason: protoc doesn't decode from plain json payload to proto object.
+//         protobuf.js supports this feature along with other protobuf functionality.
+// Usage: Under frontend/, run: `npm run build:pipeline-spec`.
+// Output: Result is stored in frontend/src/third_party/pipeline_spec.
+
+// Explaination of protobufjs cli tool:
+// Install protobufjs-cli by using the main library
+// Command: npm install protobufjs --save --save-prefix=~
+// In the future, this cli will have its own distribution, and isolate from main library.
+
+const { spawn } = require('child_process');
+
+(async () => {
+  const sourceProto = 'api/v2alpha1/pipeline_spec.proto';
+  const jsFile = 'frontend/src/third_party/pipeline_spec/pbjs_ml_pipelines.js';
+  const tsDefFile = 'frontend/src/third_party/pipeline_spec/pbjs_ml_pipelines.d.ts';
+
+  // Generate static javascript file for pipeline_spec.
+  const pbjsProcess = spawn(
+    'pbjs',
+    [`-t static-module`, `-w commonjs`, `-o ${jsFile}`, `${sourceProto}`],
```

Comment:
it seems that writing this entire script in bash would be much simpler, did you consider that? Why not?

Reply:
I agree it will be simpler to use bash shell script. But I cannot use `pbjs` cli tool directly from bash (it is part of the `protobufjs` distribution currently), so I have to use `node` in this case. Is there anything I can do to bypass this restriciton?

Reply:
why not? if not directly in bash, I think you can also just call `npx pbjs`, npx let you run commands installed in node modules.
Alternatively, you may be able to run `node_modules/protobufjs/bin/pbjs`, but you need to check.

Reply:
Anyway, I don't think these are blocking. You can put a TODO in the script too.
I'm fine with the script because it's already working.

Reply:
Thank you Yuan for the tip! I added a TODO to explain how to make the workflow simpler. I also updated the README about how to upgrade typescript definition files.